src/fp.rs

```diff
@@ -279,22 +291,9 @@ fn modp(x: u128, p: u128) -> u128 {
     z.wrapping_add(m & p)
 }
 
-pub(crate) const FP32: FieldParameters = FieldParameters {
-    p: 4293918721, // 32-bit prime
-    mu: 17302828673139736575,
-    r2: 1676699750,
-    g: 1074114499,
-    num_roots: 20,
-    bit_mask: 4294967295,
-    roots: [
-        2564090464, 1729828257, 306605458, 2294308040, 1648889905, 57098624, 2788941825,
-        2779858277, 368200145, 2760217336, 594450960, 4255832533, 1372848488, 721329415,
-        3873251478, 1134002069, 7138597, 2004587313, 2989350643, 725214187, 1074114499,
-    ],
-};
-
 pub(crate) const FP128: FieldParameters = FieldParameters {
     p: 340282366920938462946865773367900766209, // 128-bit prime
+    base: 18446744073709551616,
```

Comment:
Much more readable (base aligns with a word)
```suggestion
    base: 1 << 64,
```